### DISCUSSION TITLE
Should we move withdrawMultiple from SablierV2Lockup to batch contract?

### DISCUSSION THREAD

**Discussion by andreivladbrg** [sablier-labs/evm-monorepo#891] [Ideas]
The current `withdrawMultiple` does not take a to address parameter.

Also, we have made the `withdraw` function public, allowing anyone to call it if the to address is the recipient.

Thus, we can completely remove it from `SablierV2Lockup` to reduce the size.

We would need to call `getRecipient` for each `streamId`, which might slightly increase the gas cost. However, since the purpose of `withdrawMultiple` is not to be as optimized as possible, would it be a good idea to move it to the batch contract?

@sablier-labs/solidity

**Comment by PaulRBerg** [comment]
Interesting idea.

Have we bumped into the contract size limit with `LockupDynamic` or `LockupTranched`? If not, I think it should be fine to keep it where it is now. Moving this function to the `Batch` would not confer us (Sablier) any benefit, since we would pay the gas for deploying this logic on-chain in any case (

**Reply by smol-ninja** [reply]
> Have we bumped into the contract size limit with LockupDynamic or LockupTranched?

Not with the optimized build.

**Comment by smol-ninja** [comment]
I like the idea. It reduces the contract size by the following margins (optimized builds):

1. LockupDynamic: -300
2. LockupLinear: -1200
3. LockupTranched: -1000

However, since 
- it doesn't have any direct benefit at the moment as we are below the limits.
- it increases the gas cost of `withdrawMultiple`

I suggest to not implement it right now but we keep it in mind for the future when we hit the contract size limit.

**Reply by PaulRBerg** [reply]
Great. And I agree, it's a good idea to keep in our back pocket.

I suppose the unit for those margins are bytes?

**Reply by smol-ninja** [reply]
Yes.
